### DISCUSSION TITLE
Vaillant d.29 diagnostic code PrimaryCircuitFlowRate query

### DISCUSSION THREAD

**Discussion by JonesGW-MGD** [john30/ebusd#1153] [Q&A]
Hi all,

I have a recently installed v5 adapter and largely working HA integration due in no small part to the efforts of John and this community of course.

My objective has been to record time-series data for my new boiler  - ideally as many of the diagnostic codes and other data as possible - which is a Vaillant 835 combination boiler, sold in the UK at least as the EcoTec 835 exclusive Green IQ  835 (VUW 356/5‐7 (H-GB)) and which presents on ebus as SW=0604 & HW=9102.

I've had a lot of what looks like successes but still a few queries/issues, so thought I'd start with what seems like a discrete query.

I am particularly interested in the HC flow rate as I have some concerns about the implementation of the wet side of set-up - in particularly whether the pipe sizings is causing velocity noises etc.

I've mapped d.29 (which I 'borrowed' from bai.0010003857.inc) to successfully retrieve data:

```
r,,d.29_PrimaryCircuitFlowRate,d.29 PrimaryCircuitFlowRate_DK,,,,"FB00",,,uin100,,,primary circuit water flow rate m3/h
``` 
But two odd things - first, the results seem really high:

<img width="250" alt="image" src="https://github.com/john30/ebusd/assets/68991992/7e4ff5d3-7c99-45e6-a1a2-0fb1ee61b2a5">

15 odd m3/h is about 4 l/s

Second, d.29 on the panel shows in l/h not m3/h and comparatively shows 874 l/h or about .24 l/s

Incidentally, .24 l/s converts to between about .8 m/s to 1.5 m/s depending on the pipe size so that would be fine by UK design standards for water velocity in HC pipework - 4 l/s would very much not be fine.

in the first instance, was wondering if anyone had any thoughts about why the units might differ from the bus message to the physical panel and why they might read what seems very high!

Any thoughts welcome!

Gareth

**Comment by keynet** [comment]
My new Ecotec plus 20kW (VU 20CS/1-5 (N-GB)) shows a flow of ~1500l/h on the boiler UI when the tank is being charged, the equivalent data that comes back on ebusd is "15.04" for example, the unit is quoted as "uin100" in the config file.

I'm guessing that means multiply by 100?... which would then equal the boiler UI reading, just less than 0.5l/s. The Megaflo is close by, and on 22mm piping of course.  The pump - ebusd "PumpPower" - is reported at 100%.
Your readings are about half, but not orders of magnitude different.

When only radiators are on, the flow is 7 - 9 (15mm pipe, sounds right, about half x-section area), and 5 - 6 when the under floor only running. The latter is constrained by the manifold and the ability to consume energy, the pump drops to < 30% then, as does the modulation, and fan speed.

**Reply by JonesGW-MGD** [reply]
That's really helpful feedback, thanks!  I'll have to go back and have another play with it!

**Reply by JonesGW-MGD** [reply]
It's odd though - on my boiler d.29 from the installer panel doesn't correlate with the d.29 message in ebus - even accounting for a /100 divisor - it seems out by * 50 to  60  and not * 100 so a bit puzzled currently!

**Reply by keynet** [reply]
My numbers appear to line up in this case, but yours is a different boiler with different firmware.  The only way to be sure would be to have an independent flow measurement, rarely if ever included in a design.  Hard to think why they would use different scales for different systems though - would make for lots of incompatibility.

Similarly I'm trying to scale my energy measurements vs. boiler UI and having some issues.  I started with info from [this post](https://github.com/john30/ebusd-configuration/pull/411/commits/7a36c071b8d29b5325061367a395de679dcab092) to get figures reported (the config I have, also the default onlne one doesn't report energy for my boiler), but while it now reports numbers, the suggested scaling is equally not right, and doesn't produce believable figures.  FYI I start by assuming the sw engineer used binary scaling as that's easier and faster - probably rather old skool these days - but in this case divisors of 4096 and 4096 x 32 are producing numbers consistent with the UI. Equally could be 4000 and 30 but I'm working on the oldskool premise for the moment.  In your case I'd try 64 as a divisor.
